src/lminiz.c

```diff
@@ -427,8 +414,7 @@ static int lmz_uncompress(lua_State* L)
   unsigned char* outb;
   in_len = 0;
   inb = (const unsigned char*)luaL_checklstring(L, 1, &in_len);
-
-  out_len = in_len;
+  out_len = luaL_optint(L, 2, in_len * 10);
```

Comment:
deflate should make string shorter, so out_len = in_len * 10 is effective value?

Reply:
I think `out_len = mz_compressBound(in_len)` can be use here.

Reply:
"inflate" is "uncompress" or "decompress"; "deflate" is "compress". `mz_compressBound` is used in `lmz_compress`.

Reply:
/*   MZ_BUF_ERROR if no forward progress is possible because the input buffer is empty but the inflater needs more input to continue, or if the output buffer is not large enough. Call mz_inflate() again */
How about this.

```
  out_len = luaL_optint(L, 2, in_len * 2);
  while(1) {
  outb = malloc(out_len);
  ret = mz_uncompress(outb, &out_len, inb, in_len);
  if (ret==MZ_BUF_ERROR) {
     out_len *= 2;
     free(outb)
  }else if(ret==0)
    break;
  }
```

Reply:
Hmm. If the stream is very large, then this gets stuck in a MZ_BUF_ERROR loop.

```lua
local function test(n)
	local data = {}
	for i = 1, 2^n do
		data[i] = string.char(math.random(0, 255))
	end
	data = table.concat(data)
	print('input len:', #data)
	local c = miniz.compress(data)
	print('compressed len:', #c)
	local u = miniz.uncompress(c)
	assert(data == u)
end

test(8)
test(16)
test(32)
```